packages/api-server/src/index.ts

```diff
@@ -45,6 +47,25 @@ function safeDecorator(cb: (req: IncomingMessage) => MaybePromise<void>) {
     };
 }
 
+function createCeroServerConfig(conf: ServerConfig = {}): any {
+    if (conf.server) {
+        return conf.server;
+    }
+
+    if (conf.sslKeyPath && conf.sslCertPath) {
+        const https = require("https");
+        const fs = require("fs");
+        const sslConfig = {
+            key: fs.readFileSync(conf.sslKeyPath),
+            cert: fs.readFileSync(conf.sslCertPath)
+        };
+
+        return https.createServer(sslConfig);
+    }
+
+    return undefined;
+}
```

Comment:
```suggestion
function createCeroServerConfig(conf: ServerConfig = {}): ServerConfig["server"] {
    if (conf.server) {
        return conf.server;
    }

    if (conf.sslKeyPath && conf.sslCertPath) {
        const sslConfig = {
            key: fs.readFileSync(conf.sslKeyPath),
            cert: fs.readFileSync(conf.sslCertPath)
        };

        return createHttpsServer(sslConfig);
    }

    return createHttpServer();
}
```

Reply:
what do you think?

Reply:
Brilliant! Didn't know how to approach that type thanks! :)